docs/rfc/0696-p2p-node.adoc

```diff
@@ -0,0 +1,302 @@
+= RFC: Peer-to-peer Node
+:author: @xla
+:revdate: 2021-06-30
+:revremark: draft
+:toc:
+:toc-placement: preamble
+
+* Author: {author}
+* Date: {revdate}
+* Status: {revremark}
+* Community discussion: n/a
+* Tracking Issue: n/a
+
+== Overview
+
+This RFC expands on the peer-to-peer node outlined in <<rfc-0682, RFC 0682>> to
+propose a new stand-alone daemon with focus on driving the p2p stack, exposing a
+minimal API and defer supervision to more integrated and mature system services.
+Furthermore it will prove that all daemons are made equal and only distinct with
+regards to the configuration given (e.g. tracking behaviour) and lifetime of the
+process (i.e. socket-activated or long-running).
+
+It also strives to recommend APIs and mechanisms -- in the form of non-core
+protocols - which remove the need for online requests made by clients to a
+running node.
+
+== Terminology and Conventions
+
+The key words "`MUST`", "`MUST NOT`", "`REQUIRED`", "`SHALL`", "`SHALL NOT`",
+"`SHOULD`", "`SHOULD NOT`", "`RECOMMENDED`", "`NOT RECOMMENDED`", "`MAY`", and
+"`OPTIONAL`" in this document are to be interpreted as described in <<RFC2119>>
+and <<RFC8174>> when, and only when, they appear in all capitals, as shown here.
+
+CBOR <<RFC8949>> datatype definitions are given using the notation devised in
+CDDL <<RFC8610>>. By convention, `struct`-like datatypes are encoded as CBOR
+maps, where the map key is the zero-based numeric index of the field in
+declaration order.
+
+== Protocols
+
+The main purpose of the node is to operate the core protocol of link currently
+implemented in `librad::net::peer`, which currently consists of holding onto the
+futures du jour and dropping them at opportune moments. Furthermore all
+information in form of synchrounous calls and events obtainable internally via
+the sufficiently space-engineered tincans SHALL be made available.
+
+Besides the core protocol other functionality will be encapsulated in protocols
+which will be driven by a stream of well-defined inputs. The well-defined
+outputs of these protocols are used for effect management.
+
+=== Subroutines
+
+This section expands on the auxilliary routines driven by the node on top of the
+core protocol. Where each SHOULD allow to be disabled through configuration. A
+subroutine acts as reactor driving the protocols by feeding inputs - usually
+results of effects - into them and informed by the outputs schedules new
```

Comment:
This sentence could use some reworking. I'm not sure what you're trying to say about scheduling new effects :)

Reply:
884b5e5